src/modules/ie-objects/elasticsearch/queryBuilder.ts

```diff
@@ -197,16 +197,19 @@ export class QueryBuilder {
 		filters: SearchFilter[] | undefined,
 		inputInfo: QueryBuilderInputInfo
 	) {
-		if (!filters || isEmpty(filters)) {
-			// Return query object that will match all results
-			return { match_all: {} };
-		}
-
 		const filterObject: any = {};
 		// Add additional filters to the query object
 		const filterArray: any[] = [];
 		set(filterObject, 'bool.filter', filterArray);
 
+		// Kiosk users are only allowed to view objects from their maintainer
+		if (inputInfo.user.getGroupName() === GroupName.KIOSK_VISITOR) {
+			filterArray.push({
```

Comment:
REQ: this will never work as you don't do anything with filterArray afterwards. You should check why logic in buildLicenseFilter doesn't work as intended.

```
// KIOSK users and CP Admins always have access to the visitor space that they are linked to.
		if (user.getMaintainerId()) {
```

Reply:
Resolved after decision to keep it like this after daily.